extensions/resource-deployment/src/services/tools/toolBase.ts

```diff
@@ -2,64 +2,227 @@
+			this.logToOutputChannel(`   ${localize('toolBase.getPip3InstallationLocation.LocationNotFound', "Could not find 'Location' in the output:")}`);
+			this.logToOutputChannel(pip3ShowOutput, `   ${localize('toolBase.getPip3InstallationLocation.Output', 'output:')}`);
+			return '';
+		} else {
+			return retValue;
 		}
-		this._isInstalled = false;
-		this._statusDescription = undefined;
-		this._version = undefined;
-		this._versionOutput = undefined;
-		return this._platformService.runCommand(this.versionCommand).then((stdout) => {
-			this._versionOutput = stdout;
-			this._version = this.getVersionFromOutput(stdout);
-			if (this._version) {
-				this._isInstalled = true;
+	}
+
+	public get outputChannelName(): string {
+		return this._platformService.outputChannelName();
+	}
+
+	public showOutputChannel(preserveFocus?: boolean | undefined): void {
+		this._platformService.showOutputChannel(preserveFocus);
+	}
+
+	public async install(): Promise<void> {
+		try {
+			this.status = ToolStatus.Installing;
+			await this.installCore();
+			this.status = ToolStatus.Installed;
+			await this.checkAndUpdateVersion();
+			await this.addInstallationSearchPathsToSystemPath();
+		} catch (error) {
+			const errorMessage = getErrorMessage(error);
+			this._statusDescription = localize('toolBase.InstallError', "Error installing tool '{0}'.{1}Error: {2}{1}See output channel '{3}' for more details", this.displayName, EOL, errorMessage, this.outputChannelName);
+			this.status = ToolStatus.Error;
+			throw error;
+		}
+	}
+
+	protected async installCore() {
+		const installationCommands: Command[] | undefined = this.installationCommands;
+		if (!installationCommands || installationCommands.length === 0) {
+			throw new Error(localize('toolBase.installCore.CannotInstallTool', 'Cannot install tool:${0}::${1} as installation commands are unknown', this.displayName, this.description));
+		}
+		for (let i: number = 0; i < installationCommands.length; i++) {
+			await this._platformService.runCommand(installationCommands[i].command,
+				{
+					workingDirectory: installationCommands[i].workingDirectory || this.downloadPath,
+					additionalEnvironmentVariables: installationCommands[i].additionalEnvironmentVariables,
+					sudo: installationCommands[i].sudo,
+					commandTitle: installationCommands[i].comment,
+					ignoreError: installationCommands[i].ignoreError
+				},
+			);
+		}
+	}
+
+	protected async addInstallationSearchPathsToSystemPath(): Promise<void> {
+		const installationPath = await this.getInstallationPath();
+		const searchPaths = [installationPath, ...this.installationSearchPaths].filter(path => !!path);
+		this.logToOutputChannel(`${localize('toolBase.addInstallationSearchPathsToSystemPath.SearchPaths', 'Search Paths for tool')} '${this.displayName}': ${JSON.stringify(searchPaths, undefined, '\t')}`); //this.diplayName is localized and searchPaths are OS filesystem paths.
```

Comment:
don't think this needs to be localized.

Reply:
we are outputing to outputChannel which is user visible so any messaging for the actualy paths should be localized.

Reply:
Fixing to make srue I use a double quoted literal string for 2nd parameter of localize method

Reply:
fixed formation, but keeping them as localized, actual paths are not localized so we as devs will be able to follow this.